### PR TITLE
WT-8420 FLCS support for test_checkpoint

### PR DESCRIPTION
**Pull request by sauclovian-wt**
I've added FLCS support to test_checkpoint.

See the first changeset's change message for the gory details.

The second changeset does a cosmetic reorganization of the options in the test lists; if people don't like it I'm fine with dropping it, but I think it improves the legibility a good deal.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by keithbostic** [comment]
Failed build
```[2021/11/17 20:03:58.906] Making all in test/checkpoint
[2021/11/17 20:03:58.906] make[2]: Entering directory '/data/mci/3b2e3c4f873c0e774702bb116ee1cc31/wiredtiger/build_posix/test/checkpoint'
[2021/11/17 20:03:59.558]   CC       checkpointer.o
[2021/11/17 20:03:59.558]   CC       workers.o
[2021/11/17 20:03:59.558]   CC       test_checkpoint.o
[2021/11/17 20:03:59.558] ../../../test/checkpoint/test_checkpoint.c:507:40: error: implicit conversion loses integer precision: 'unsigned long' to 'u_int' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
[2021/11/17 20:03:59.558]             offset = entries[i].offset + j;
[2021/11/17 20:03:59.558]                    ~ ~~~~~~~~~~~~~~~~~~^~~
[2021/11/17 20:03:59.558] ../../../test/checkpoint/test_checkpoint.c:521:43: error: implicit conversion loses integer precision: 'unsigned long' to 'u_int' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
[2021/11/17 20:03:59.558]         offset = offset - entries[i].size + entries[i].data.size;
[2021/11/17 20:03:59.558]                ~ ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
[2021/11/17 20:03:59.558] ../../../test/checkpoint/test_checkpoint.c:508:21: error: variable 'newdigit' may be uninitialized when used here [-Werror,-Wconditional-uninitialized]
[2021/11/17 20:03:59.559]             digit = newdigit;
[2021/11/17 20:03:59.559]                     ^~~~~~~~
[2021/11/17 20:03:59.559] ../../../test/checkpoint/test_checkpoint.c:483:25: note: initialize the variable 'newdigit' to silence this warning
[2021/11/17 20:03:59.559]     char digit, newdigit;
[2021/11/17 20:03:59.559]                         ^
[2021/11/17 20:03:59.559]                          = '\0'
[2021/11/17 20:03:59.566] 3 errors generated.
[2021/11/17 20:03:59.566] make[2]: *** [Makefile:634: test_checkpoint.o] Error 1```

**Comment by keithbostic** [comment]
Reviewers: the only failure in the WT-8420 PR is `test_hs11`, so I think this one is ready to merge.

**Comment by sauclovian-wt** [comment]
It appears to be an instance of WT-8362.

File: test/checkpoint/checkpointer.c [hunk 1]
```diff
@@ -349,40 +418,85 @@ verify_consistency(WT_SESSION *session, wt_timestamp_t verify_ts)
...
     if (cursor1->get_key(cursor1, &key1) != 0 || cursor2->get_key(cursor2, &key2) != 0)
         return (log_print_err("Error getting keys", EINVAL, 1));
 
-    if (cursor1->get_value(cursor1, &val1) != 0 || cursor2->get_value(cursor2, &val2) != 0)
-        return (log_print_err("Error getting values", EINVAL, 1));
+    /*
+     * Get the values. For all table types set both the string value (so we can print) and the FLCS
+     * value.
+     */
+
```
Comment (quchenhao) [inline test/checkpoint/checkpointer.c:440]: Empty line

**Review by quchenhao — COMMENTED** [review]
The change makes sense to me. Should we also add it to data validation recovery tests? It is also running test checkpoint.

**Comment by kommiharibabu** [comment]
@quchenhao The current data validation test doesn't consider the column store also. Supporting for column store can be part of the separate ticket. @sauclovian-wt don't have access to the evergreen test results, so it would be better to do that change by any one of our team members.

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by quchenhao** [comment]
> @quchenhao The current data validation test doesn't consider the column store also. Supporting for column store can be part of the separate ticket. @sauclovian-wt don't have access to the evergreen test results, so it would be better to do that change by any one of our team members.

ok, understood.

**Review by kommiharibabu — APPROVED** [review]
LGTM With test update remaining.

File: test/checkpoint/smoke.sh [hunk 2]
```diff
@@ -7,53 +7,81 @@ test "$TESTUTIL_BYPASS_VALGRIND" = "1" && exit 0
...
+echo "checkpoint: 6 fixed-length column-store tables, named checkpoint"
+$TEST_WRAPPER ./t -t f -T 6 -c 'TeSt'
+
+echo "checkpoint: 6 fixed-length column-store tables with prepare"
+$TEST_WRAPPER ./t -t f -T 6 -p
+
+echo "checkpoint: 6 fixed-length column-store tables, named checkpoint with prepare"
+$TEST_WRAPPER ./t -t f -T 6  -c 'TeSt' -p
+
+echo "checkpoint: fixed-length column-store tables, stress history store. Sweep and timestamps"
```
[the remark this answers is not in the document]
Comment (sauclovian-wt) [reply test/checkpoint/smoke.sh:38]: Oops. Wonder how that slipped by... fixed, thanks.

**Comment by sauclovian-wt** [comment]
I created a new ticket WT-8445 for the recovery test so it doesn't get forgotten.

**Pull request merged**